test/functional/mempool_persist.py

```diff
@@ -85,5 +91,27 @@ def run_test(self):
         self.nodes.append(self.start_node(0, self.options.tmpdir))
         wait_until(lambda: len(self.nodes[0].getrawmempool()) == 5)
 
+        mempooldat0 = os.path.join(self.options.tmpdir, 'node0', 'regtest', 'mempool.dat')
+        mempooldat1 = os.path.join(self.options.tmpdir, 'node1', 'regtest', 'mempool.dat')
+        self.log.debug("Remove the mempool.dat file. Verify that dumpmempool to disk via RPC re-creates it")
+        os.remove(mempooldat0)
+        assert self.nodes[0].dumpmempool()
+        assert os.path.isfile(mempooldat0)
+
+        self.log.debug("Stop nodes, make node1 use mempool.dat from node0. Verify it has 5 transactions")
```

Comment:
I think the intent of this section of the tests is to verify that the dumped mempool from the `dumpmempool` RPC contains all the transactions. Correct? The way it's written right now doesn't test this. The act of stopping node0 causes it to dump its mempool on shutdown, and hence this test is just a repeat of the test on line 77. I think you need to:

- run `dumpmempool` on a running node
- move that mempool.dat to a temp location
- shutdown the node
- move the mempool.dat back to the correct location
- start the node
- check that its mempool contains the correct transactions